index/server/main.go

```diff
@@ -127,6 +147,21 @@ func main() {
 	router.GET("/index", serveDevfileIndex)
 	router.GET("/index.json", serveDevfileIndex)
 
+	router.GET("/index/:type", func(c *gin.Context) {
```

Comment:
This approach all work fine, but by keeping the `/index` endpoint serving _all_ stack types, it means that odo (which is currently consuming the registry library) needs to be updated to a newer version of the registry library that calls one of these endpoints instead. It also means that we can't update the hosted registry to add the samples until odo is using the stacks-only endpoint.

Would it be better to have `/index` just return stacks and have a separate endpoint for returning stacks and samples?

Reply:
Looks good to me. I’m fine with either approach, so let’s see what Elson suggests.

Reply:
Discussed this in today's cabal call, I will implement the new REST API design above: https://github.com/devfile/registry-support/pull/52#discussion_r629482807

Reply:
Updated